src/Microsoft.AspNetCore.WebUtilities/Base64UrlTextEncoder.cs

```diff
@@ -32,38 +32,30 @@ public static byte[] Decode(string text)
             return Convert.FromBase64String(DecodeToBase64String(text));
         }
 
+        // Since the string instnace passed here is owned by the Encode method, its safe to modify
+        // that instance's state, whereas in case of Decode method the passed in string instance is not owned
+        // by the Decode method, so we shouldn't be modifying it.
         // To enable unit testing
-        internal static string EncodeInternal(string base64EncodedString)
+        internal static unsafe void EncodeInternal(string base64String)
         {
-            var length = base64EncodedString.Length;
-            while (length > 0 && base64EncodedString[length - 1] == '=')
+            fixed (char* destination = base64String)
             {
-                length--;
-            }
-
-            if (length == 0)
-            {
-                return string.Empty;
-            }
-
-            var inplaceStringBuilder = new InplaceStringBuilder(length);
-            for (var i = 0; i < length; i++)
-            {
-                if (base64EncodedString[i] == '+')
-                {
-                    inplaceStringBuilder.Append('-');
-                }
-                else if (base64EncodedString[i] == '/')
-                {
-                    inplaceStringBuilder.Append('_');
-                }
-                else
+                for (var i = 0; i < base64String.Length; i++)
                 {
-                    inplaceStringBuilder.Append(base64EncodedString[i]);
+                    if (destination[i] == '+')
+                    {
+                        destination[i] = '-';
+                    }
+                    else if (destination[i] == '/')
+                    {
+                        destination[i] = '_';
+                    }
+                    else if (destination[i] == '=')
+                    {
+                        destination[i] = '.';
```

Comment:
This is because `=` character gets url encoded and also if we were to have similar behavior like earlier we would need to allocate a new string (like InplaceStringBuilder did earlier internally)